### ISSUE TITLE
Extension issue

### ISSUE THREAD

**Issue by Parmjeetjailwal** [hashicorp/vscode-terraform#805]
- Issue Type: `Bug`
- Extension Name: `terraform`
- Extension Version: `2.15.0`
- OS Version: `Windows_NT x64 10.0.19043`
- VS Code version: `1.61.0`
C:\Users\mrpar\Videos\2021-10-12 06_58_41-Microsoft Store.png 
```json 
{
	"messages": [],
	"activationTimes": {
		"codeLoadingTime": 51742,
		"activateCallTime": 57,
		"activateResolvedTime": 18381,
		"activationReason": {
			"startup": true,
			"extensionId": {
				"value": "hashicorp.terraform",
				"_lower": "hashicorp.terraform"
			},
			"activationEvent": "workspaceContainsTimeout:**/*.tf"
		}
	},
	"runtimeErrors": [
		{
			"name": "TypeError",
			"message": "Cannot read property 'document' of undefined"
		}
	],
	"runningLocation": 1
}
```
:warning: We have written the needed data into your clipboard. Please paste! :warni
![2021-10-12 06_58_41-Microsoft Store](https://user-images.githubusercontent.com/91946910/136875724-0b8ebe16-b919-42f6-9208-726543c3686a.png)
ng:

**Comment by Parmjeetjailwal** [comment]
Unresponsive

**Comment by KrustyHack** [comment]
Ay,

I've tested again this time by opening vscode directly in my terraform folder and it seems working fine on my side...

Eg. :

`(base) krustyhack@KrustyMSI:~/Development/MyProject/$ code .` => Terraform extensions is broken
`(base) krustyhack@KrustyMSI:~/Development/MyProject/$ code terraform/` => Terraform extension works fine

Looks like vscode must be started in the directory where TF files live.

**Comment by GreeFine** [comment]
I also have this issue
```bash
> code --version
1.62.0
b3318bc0524af3d74034b8bb8a64df0ccf35549a
x64
```

The logs from vscode:
```
workbench.desktop.main.js:77 [Extension Host] rejected promise not handled within 1 second: TypeError: Cannot read property 'document' of undefined
E @ workbench.desktop.main.js:77
$logExtensionHostMessage @ workbench.desktop.main.js:1525
_doInvokeHandler @ workbench.desktop.main.js:1525
_invokeHandler @ workbench.desktop.main.js:1525
_receiveRequest @ workbench.desktop.main.js:1525
_receiveOneMessage @ workbench.desktop.main.js:1525
(anonymous) @ workbench.desktop.main.js:1525
fire @ workbench.desktop.main.js:69
fire @ workbench.desktop.main.js:85
_receiveMessage @ workbench.desktop.main.js:85
(anonymous) @ workbench.desktop.main.js:85
fire @ workbench.desktop.main.js:69
acceptChunk @ workbench.desktop.main.js:85
(anonymous) @ workbench.desktop.main.js:85
k @ workbench.desktop.main.js:2967
emit @ events.js:315
addChunk @ internal/streams/readable.js:309
readableAddChunk @ internal/streams/readable.js:284
Readable.push @ internal/streams/readable.js:223
onStreamRead @ internal/stream_base_commons.js:188
workbench.desktop.main.js:77 [Extension Host] stack trace: TypeError: Cannot read property 'document' of undefined	at /home/greefine/.vscode-oss/extensions/hashicorp.terraform-2.16.0/out/extension.js:105:55	at Generator.next (<anonymous>)	at /home/greefine/.vscode-oss/extensions/hashicorp.terraform-2.16.0/out/extension.js:8:71	at new Promise (<anonymous>)	at __awaiter (/home/greefine/.vscode-oss/extensions/hashicorp.terraform-2.16.0/out/extension.js:4:12)	at /home/greefine/.vscode-oss/extensions/hashicorp.terraform-2.16.0/out/extension.js:103:64	at l.fire (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)	at /usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:95:63024	at l.fire (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)	at b.acceptDocumentsAndEditorsDelta (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:94:155958)	at b.$acceptDocumentsAndEditorsDelta (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:94:154149)	at n._doInvokeHandler (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:13802)	at n._invokeHandler (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:13486)	at n._receiveRequest (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:12148)	at n._receiveOneMessage (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:10826)	at /usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:8922	at l.fire (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)	at s.fire (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:15990)	at /usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:113:30625	at l.fire (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)	at s.fire (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:15990)	at f._receiveMessage (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:21260)	at /usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:18138	at l.fire (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)	at w.acceptChunk (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:13351)	at /usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:12699	at Socket.m (/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:113:13575)	at Socket.emit (events.js:315:20)	at addChunk (internal/streams/readable.js:309:12)	at readableAddChunk (internal/streams/readable.js:284:9)	at Socket.Readable.push (internal/streams/readable.js:223:10)	at Pipe.onStreamRead (internal/stream_base_commons.js:188:23)
E @ workbench.desktop.main.js:77
$logExtensionHostMessage @ workbench.desktop.main.js:1525
_doInvokeHandler @ workbench.desktop.main.js:1525
_invokeHandler @ workbench.desktop.main.js:1525
_receiveRequest @ workbench.desktop.main.js:1525
_receiveOneMessage @ workbench.desktop.main.js:1525
(anonymous) @ workbench.desktop.main.js:1525
fire @ workbench.desktop.main.js:69
fire @ workbench.desktop.main.js:85
_receiveMessage @ workbench.desktop.main.js:85
(anonymous) @ workbench.desktop.main.js:85
fire @ workbench.desktop.main.js:69
acceptChunk @ workbench.desktop.main.js:85
(anonymous) @ workbench.desktop.main.js:85
k @ workbench.desktop.main.js:2967
emit @ events.js:315
addChunk @ internal/streams/readable.js:309
readableAddChunk @ internal/streams/readable.js:284
Readable.push @ internal/streams/readable.js:223
onStreamRead @ internal/stream_base_commons.js:188
workbench.desktop.main.js:1851 [[object Object]]Cannot read property 'document' of undefined
$onExtensionRuntimeError @ workbench.desktop.main.js:1851
_doInvokeHandler @ workbench.desktop.main.js:1525
_invokeHandler @ workbench.desktop.main.js:1525
_receiveRequest @ workbench.desktop.main.js:1525
_receiveOneMessage @ workbench.desktop.main.js:1525
(anonymous) @ workbench.desktop.main.js:1525
fire @ workbench.desktop.main.js:69
fire @ workbench.desktop.main.js:85
_receiveMessage @ workbench.desktop.main.js:85
(anonymous) @ workbench.desktop.main.js:85
fire @ workbench.desktop.main.js:69
acceptChunk @ workbench.desktop.main.js:85
(anonymous) @ workbench.desktop.main.js:85
k @ workbench.desktop.main.js:2967
emit @ events.js:315
addChunk @ internal/streams/readable.js:309
readableAddChunk @ internal/streams/readable.js:284
Readable.push @ internal/streams/readable.js:223
onStreamRead @ internal/stream_base_commons.js:188
workbench.desktop.main.js:1851 TypeError: Cannot read property 'document' of undefined
	at vscode-file://vscode-app/home/greefine/.vscode-oss/extensions/hashicorp.terraform-2.16.0/out/extension.js:105
	at Generator.next (<anonymous>)
	at vscode-file://vscode-app/home/greefine/.vscode-oss/extensions/hashicorp.terraform-2.16.0/out/extension.js:8
	at new Promise (<anonymous>)
	at __awaiter (vscode-file://vscode-app/home/greefine/.vscode-oss/extensions/hashicorp.terraform-2.16.0/out/extension.js:4)
	at vscode-file://vscode-app/home/greefine/.vscode-oss/extensions/hashicorp.terraform-2.16.0/out/extension.js:103
	at l.fire (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66)
	at vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:95
	at l.fire (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66)
	at b.acceptDocumentsAndEditorsDelta (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:94)
	at b.$acceptDocumentsAndEditorsDelta (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:94)
	at n._doInvokeHandler (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98)
	at n._invokeHandler (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98)
	at n._receiveRequest (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98)
	at n._receiveOneMessage (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98)
	at vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98
	at l.fire (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66)
	at s.fire (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74)
	at vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:113
	at l.fire (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66)
	at s.fire (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74)
	at f._receiveMessage (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74)
	at vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74
	at l.fire (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66)
	at w.acceptChunk (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74)
	at vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74
	at Socket.m (vscode-file://vscode-app/usr/lib/code/out/vs/workbench/services/extensions/node/extensionHostProcess.js:113)
	at Socket.emit (events.js:315)
	at addChunk (internal/streams/readable.js:309)
	at readableAddChunk (internal/streams/readable.js:284)
	at Socket.Readable.push (internal/streams/readable.js:223)
	at Pipe.onStreamRead (internal/stream_base_commons.js:188)
$onExtensionRuntimeError @ workbench.desktop.main.js:1851
_doInvokeHandler @ workbench.desktop.main.js:1525
_invokeHandler @ workbench.desktop.main.js:1525
_receiveRequest @ workbench.desktop.main.js:1525
_receiveOneMessage @ workbench.desktop.main.js:1525
(anonymous) @ workbench.desktop.main.js:1525
fire @ workbench.desktop.main.js:69
fire @ workbench.desktop.main.js:85
_receiveMessage @ workbench.desktop.main.js:85
(anonymous) @ workbench.desktop.main.js:85
fire @ workbench.desktop.main.js:69
acceptChunk @ workbench.desktop.main.js:85
(anonymous) @ workbench.desktop.main.js:85
k @ workbench.desktop.main.js:2967
emit @ events.js:315
addChunk @ internal/streams/readable.js:309
readableAddChunk @ internal/streams/readable.js:284
Readable.push @ internal/streams/readable.js:223
onStreamRead @ internal/stream_base_commons.js:188
workbench.desktop.main.js:77 [Extension Host] Active Window Changed
```

**Comment by dbanck** [comment]
Thanks a lot for the stack trace, @GreeFine.

The good news is, we fixed the error you posted within the upcoming 2.17.0 release of the extension.

The bad news is that we couldn't cut a release for 2.17.0 yet; it will take us a few more days.

**Comment by jpogran** [comment]
@Parmjeetjailwal @GreeFine We released v2.17.0 on 2021/12/02 that contained several fixes surrounding issues you describe. This should reduce the errors that can occur when files are open before the extension is fully activated. We have a [preview extension](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform-preview) that has additional work that we haven't released to [stable](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform) yet that addresses opening untitled documents if you are interested in testing that out.

@Parmjeetjailwal the terminal pty host process error is unrelated to the terraform extension, and it may have already been solved by recent VS Code updates where they were working on resuming suspended terminal sessions.

@KrustyHack In addition to using the update described above, you should read my update in issue [#542](https://github.com/hashicorp/vscode-terraform/issues/542#issuecomment-1000401937) where I outlined some information about how to configure automatic formatting in VS Code. I've ported the relevant parts of my explanation to PR #890 which will update the README with this information. Be sure to read that issue or PR because it outlines information on how to successfully configure the extension to format files. Hopefully that content will help you, if not please open a separate new issue.

Thank you everyone for contributing information to this thread. This is actively being investigated, but it is hard to investigate the problem if no new information is provided. Adding a +1 reaction is helpful in gauging how many people have seen this issue, but not helpful with giving us diagnostic information we need to figure out what is going on.

With the variety of configurations that are possible, it's hard to address each person in a single thread when this is likely situation dependent. We can address each person on an individual basis in individual tickets, where it is much easier to ask each person for needed diagnostic information compared to one long ticket.

Since this has been open for a long time, I'm going to lock and close this issue now. This is not an indication we are ignoring the issue, but rather an attempt to gauge how recent an issue it is and to gain new diagnostic information. Please do open a new issue if problems persist for you.
